packages/x-date-pickers/src/AdapterDateFns/index.ts

```diff
@@ -16,6 +16,27 @@ const formatTokenMap: MuiFormatTokenMap = {
   MMM: { sectionName: 'month', contentType: 'letter' },
   LLL: { sectionName: 'month', contentType: 'letter' },
   LLLL: { sectionName: 'month', contentType: 'letter' },
+  E: { sectionName: 'weekDay', contentType: 'letter' },
+  EE: { sectionName: 'weekDay', contentType: 'letter' },
+  EEE: { sectionName: 'weekDay', contentType: 'letter' },
+  EEEE: { sectionName: 'weekDay', contentType: 'letter' },
+  EEEEE: { sectionName: 'weekDay', contentType: 'letter' },
```

Comment:
Do you mean that we can't access `Thursday` with the keyboard when using the `E` format ?
If so, I think it's something that we can expect, it remains accessible with the digits and the arrow.
Tbh, some of these formats will probably never be used for editing :grimacing:

Reply:
Yeah, I meant exactly that. 👌 
Sure, I'd assume the same, that they are going to be used very rarely and shouldn't be a big problem.
But in a very rare case we could probably come up with a fix supporting keyboard entry autocompletion for longer length than the section, in case there are more than 1 found items.
Because we'd have the same issue with other "exotic" format tokens, like [`MMMMM`](https://date-fns.org/v2.29.3/docs/format#:~:text=2-,MMMMM,-J%2C%20F%2C%20...%2C%20D). 🙈

Reply:
How do you suggest we could support it ?

Reply:
I'm honestly not suggesting to go ahead and rush introducing this extra complexity until there is actual need for it.
If we'd proceed with handling every rare condition, our fields codebase and it's complexity might explode. 🙈 

But if we'd need support for it. I believe it shouldn't be too difficult allowing letter section auto-complete regardless of the section format length. If we still have > 1 possible autocomplete values based on the input—allow section autocomplete to work with more than 1 character.
For example:
If we mentioned one letter week day format:
- Entering `t` shows `T` in the section, but does not jump to the next section, because there are > 1 options
- Entering `h` still shows only `T` in the section, but does jump to the next section, because there is only one weekday option for `th` search string

Reply:
OK
The problem here is that we are doing the auto-completion on the value formatted with the current format. We don't have the `h` of `Thursday`.
One workaround could be that if we end up with several identical values matching the query, we fallback to a full week day format.
Another would be to always use the full week day format, but that's a change that would need to be carefully checked (could we have some format in some locale that are not just shortcuts of the full format and that would be hard to use if we autocomplete based on the full format and render those weird formats ?)

Maybe we can wait for feedback on this one :+1:

Reply:
> The problem here is that we are doing the auto-completion on the value formatted with the current format. We don't have the h of Thursday.

Oh, fair point. 😞 

> One workaround could be that if we end up with several identical values matching the query, we fallback to a full week day format.

> Maybe we can wait for feedback on this one 

Indeed, based on this discussion findings it seems like the added complexity is definitely not worth it until there is enough incentive from users. 🙈